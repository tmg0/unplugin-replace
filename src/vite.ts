/**
 * This entry file is for Vite plugin.
 *
 * @module
 */

import unplugin from './index'

/**
 * Vite plugin
 *
 * @example
 * ```ts
 * // vite.config.ts
 * import Replace from 'unplugin-replace/vite'
 *
 * export default defineConfig({
 *   plugins: [Replace()],
 * })
 * ```
 */
export default unplugin.vite
