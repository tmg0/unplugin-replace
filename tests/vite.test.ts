import path from 'node:path'
import { expect, test } from 'vitest'
import { build } from 'vite'
import UnpluginReplace from '../src/vite'
import type { RollupOutput } from 'rollup'

test('vite', async () => {
  const root = path.resolve(__dirname, 'fixtures')
  const { output } = (await build({
    root,
    build: {
      minify: false,
      rollupOptions: {
        input: [path.resolve(root, 'main.js')],
        external: (id) => id === 'node:process',
        logLevel: 'silent',
      },
      write: false,
    },
    logLevel: 'silent',
    plugins: [
      UnpluginReplace({
        'process.platform': '"darwin"',
      }),
    ],
  })) as RollupOutput
  expect(output[0].code).toMatchSnapshot()
})
