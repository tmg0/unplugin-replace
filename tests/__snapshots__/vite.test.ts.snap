// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite 1`] = `
"import "node:process";
const platform = "darwin";
console.log('hello "darwin"', platform);
console.log(DEV);
"
`;
